app/controllers/admin/articles_controller.rb

```diff
@@ -90,10 +89,11 @@ def set_article
     end
 
     def set_parent_categories
-      return unless params[:id].present?
-      @category = Category.friendly.find(params[:id])
-      @subcategories = @category.subcategories
-      @teams = @subcategories.flat_map{ |subcat| subcat.teams}
+      return unless params[:id] or params[:category_slug]
+      category_slug = params[:category_slug].present? ? params[:category_slug] : params[:id]
+      @category = Category.friendly.find(category_slug)
+      @subcategories = @category.subcategories || []
+      @teams = @subcategories.flat_map{ |subcat| subcat.teams} || []
```

Comment:
instead of `@subcategories.flat_map{ |subcat| subcat.teams}` it could be `@subcategories.flat_map(&:teams)`